# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wgpu_bindings/src/identity.rs
# Commit: ef5dc3e04e5f
# Full Hash: ef5dc3e04e5f271eea0636ab3a495e95cc912f1d
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-09-04 09:40:24
# Regressor Bug: 1726626
# File Overlap Count: 3
# Description:
#   Bug 1726626 - Move gfx/wgpu into a 3rd party dependency r=jgilbert,bholley
#   
#   This update makes wgpu a vendored dependency instead of having it in gfx/wgpu.
#   
#   ## Notes
# ==============================================================================

diff --git a/gfx/wgpu_bindings/src/identity.rs b/gfx/wgpu_bindings/src/identity.rs
--- a/gfx/wgpu_bindings/src/identity.rs
+++ b/gfx/wgpu_bindings/src/identity.rs
@@ -29,17 +29,16 @@ impl<I: id::TypedId + Clone + std::fmt::
 }
 
 //TODO: remove this in favor of `DropAction` that could be sent over IPC.
 #[repr(C)]
 pub struct IdentityRecyclerFactory {
     param: FactoryParam,
     free_adapter: extern "C" fn(id::AdapterId, FactoryParam),
     free_device: extern "C" fn(id::DeviceId, FactoryParam),
-    free_swap_chain: extern "C" fn(id::SwapChainId, FactoryParam),
     free_pipeline_layout: extern "C" fn(id::PipelineLayoutId, FactoryParam),
     free_shader_module: extern "C" fn(id::ShaderModuleId, FactoryParam),
     free_bind_group_layout: extern "C" fn(id::BindGroupLayoutId, FactoryParam),
     free_bind_group: extern "C" fn(id::BindGroupId, FactoryParam),
     free_command_buffer: extern "C" fn(id::CommandBufferId, FactoryParam),
     free_render_bundle: extern "C" fn(id::RenderBundleId, FactoryParam),
     free_render_pipeline: extern "C" fn(id::RenderPipelineId, FactoryParam),
     free_compute_pipeline: extern "C" fn(id::ComputePipelineId, FactoryParam),
@@ -66,26 +65,16 @@ impl wgc::hub::IdentityHandlerFactory<id
     fn spawn(&self, _min_index: u32) -> Self::Filter {
         IdentityRecycler {
             fun: self.free_device,
             param: self.param,
             kind: "device",
         }
     }
 }
-impl wgc::hub::IdentityHandlerFactory<id::SwapChainId> for IdentityRecyclerFactory {
-    type Filter = IdentityRecycler<id::SwapChainId>;
-    fn spawn(&self, _min_index: u32) -> Self::Filter {
-        IdentityRecycler {
-            fun: self.free_swap_chain,
-            param: self.param,
-            kind: "swap_chain",
-        }
-    }
-}
 impl wgc::hub::IdentityHandlerFactory<id::PipelineLayoutId> for IdentityRecyclerFactory {
     type Filter = IdentityRecycler<id::PipelineLayoutId>;
     fn spawn(&self, _min_index: u32) -> Self::Filter {
         IdentityRecycler {
             fun: self.free_pipeline_layout,
             param: self.param,
             kind: "pipeline_layout",
         }